### ISSUE TITLE
"Show Deleted" checkbox in /admin/products doesn't stay checked when appropriate

### ISSUE THREAD

**Issue by spaghetticode** [solidusio/solidus#1903]
When filtering products in the admin page the checkbox `Show Deleted`, if checked, doesn't remain checked after form submission. 

This causes also a problem with pagination, you can't reach any other page expect the first with deleted products.

**Comment by spaghetticode** [comment]
The reason is due to the following code in `backend/app/controllers/spree/admin/products_controller.rb:105`: 
```
@collection = @collection.with_deleted if params[:q].delete(:deleted_at_null) == '0'
```
I would change it as follow: 
```
@collection = @collection.with_deleted if params[:q][:deleted_at_null] == '0'
```
before opening a new PR with my changes I would like to ask you if there's a good reason for deleting `deleted_at_null` from `params[:q]` in that line. Thanks!

**Comment by mtomov** [comment]
https://github.com/solidusio/solidus/commit/de2065b seems to be the commit to introduce the `delete()`. I guess if you are up for writing a test for this behaviour, there's nothing to stop you from making that change.

Actually, isn't it possible to directly use the `ransackable_scopes` to allow this `with_deleted` https://github.com/activerecord-hackery/ransack#using-scopesclass-methods

**Comment by spaghetticode** [comment]
@mtomov I'm not sure I completely understood your last sentence. Did you mean that, instead of using `deleted_at_null` we may use a (new?) scope on the model, which we also use as a ransackable scope, and then we also change the param name on the `erb` view?

Anyway, I choose the simplest path for my PR: remove the `delete` method call. I also added a simple controller spec for it.

**Comment by spaghetticode** [comment]
@mtomov with a little help from @kennyadsl I think I finally got your point. I am working on a new implementation that uses ransackable scopes, which has the extra benefit of tidying up a bit the code in the controller. I think that's an improvement over my first version, so I will update my PR ASAP. But if you don't prefer the new version then I can still go back to the first proposal ;)

**Comment by mtomov** [comment]
Sure, take your time.

I'm just not sure if you will be able to capture the existing bug of the "not-checked" checkbox after searching with a `controller` test. Maybe a `feature` test might do one better.

**Comment by spaghetticode** [comment]
Thanks for the suggestion, I've just added also an expectation in an existing feature spec as well.

**Comment by mamhoff** [comment]
Fixed via #1904
